File: 0795-Number of Subarrays with Bounded Maximum/0795-Number of Subarrays with Bounded Maximum.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int>& A, int L, int R) {
        int count = 0;
        int maxIndex = -1, minIndex = -1;
        for (int i = 0; i < A.size(); ++i) {
            if (A[i] >= L) {
                minIndex = i;
            }
            if (A[i] > R) {
                maxIndex = i;
            }
            count += minIndex - maxIndex;
        }
        return count;
    }
};

File: 0701-0800/0795-Number of Subarrays with Bounded Maximum/0795-Number of Subarrays with Bounded Maximum.py
class Solution:
    def numSubarrayBoundedMax(self, A: List[int], L: int, R: int) -> int:
        count = 0
        left = right = -1
        for i in range(len(A)):
            if A[i] >= L:
                right = i
            if A[i] > R:
                left = i
            count += right - left
        return count
